ci-operator/step-registry/upi/install/gcp/by-infra-manager/upi-install-gcp-by-infra-manager-commands.sh

```
@@ -0,0 +1,729 @@
+export KUBECONFIG
+CLUSTER_NAME="$(jq -r .clusterName metadata.json)"
+INFRA_ID="$(jq -r .infraID metadata.json)"
+PROJECT_NAME="$(jq -r .gcp.projectID metadata.json)"
+REGION="$(jq -r .gcp.region metadata.json)"
+
+## Available zones and instance zones might be different in region for arm64 machines
+mapfile -t AVAILABILITY_ZONES < <(gcloud compute regions describe "${REGION}" --format=json | jq -r '.zones[]' | cut -d "/" -f9)
+mapfile -t MASTER_INSTANCE_ZONES < <(gcloud compute machine-types list --filter="zone:(${REGION}) AND name=(${CONTROL_PLANE_NODE_TYPE})" --format=json | jq -r '.[].zone')
+mapfile -t MASTER_ZONES < <(echo "${AVAILABILITY_ZONES[@]}" "${MASTER_INSTANCE_ZONES[@]}" | sed 's/ /\n/g' | sort -R | uniq -d)
+for index in {0..2}; do
+  eval ZONE_${index}=${MASTER_ZONES[index]}
```

Shellcheck (0.11.0)

[info] Double quote to prevent globbing and word splitting.

(SC2086)